pymeasure/instruments/feeltech/fy3200s.py

```diff
@@ -0,0 +1,314 @@
+        validator=step_validator(1),
+        values=(0, 99),
+        get_process=lambda x: float(x.strip("ct")),
+    )
+
+    sweep_beginning_frequency = Instrument.setting(
+        "{ch}b%.9i",
+        """Set the sweep beginning frequency (Hz)""",
+        validator=step_validator(1),
+        values=(0, 100000000),
+        set_process=lambda x: x * 100,
+    )
+
+    sweep_end_frequency = Instrument.setting(
+        "{ch}e%.9i",
+        """Set the sweep beginning frequency (Hz)""",
+        validator=step_validator(1),
+        values=(0, 100000000),
+        set_process=lambda x: x * 100,
+    )
+
+    sweep_mode = Instrument.setting(
+        "{ch}m%i",
+        "Set the sweepmode (LINEAR or LOG).",
+        validator=strict_discrete_set,
+        map_values=True,
+        values=SWEEPMODES,
+    )
+
+    def start_sweep(self):
+        self.write("{ch}r1")
+
+    def pause_sweep(self):
+        self.write("{ch}r0")
+
+
+class SubsidiaryChannel(Channel):
+    WAVEFORMS = {
+        "SINE": 0,
+        "SQUARE": 1,
+        "TRIANGLE": 2,
+        "SAWTOOTH_RISE": 3,
+        "SAWTOOTH_FALL": 4,
+        "DC": 5,
+        "PULSE_LORENTZ": 6,
+        "MULTITONE": 7,
+        "RANDOM_NOISE": 8,
+        "ELECTROCARDIOGRAM": 9,
+        "PULSE_TRAPEZ": 10,
+        "PULSE_SINC": 11,
+        "PULSE_NARROW": 12,
+        "GAUSSIAN_NOISE": 13,
+        "AMP_MOD": 14,
+        "FREQ_MOD": 15,
+        "ARB1": 16,
+        "ARB2": 17,
+        "ARB3": 18,
+    }
+
+    waveform = Instrument.setting(
+        "{ch}w%i",
+        """Set the waveform of the channel.
+
+        The waveform must be one of the following:
+
+        - SINE
+        - SQUARE
+        - TRIANGLE
+        - SAWTOOTH_RISE
+        - SAWTOOTH_FALL
+        - DC
+        - PULSE_LORENTZ
+        - MULTITONE
+        - RANDOM_NOISE
+        - ELECTROCARDIOGRAM
+        - PULSE_TRAPEZ
+        - PULSE_SINC
+        - PULSE_NARROW
+        - GAUSSIAN_NOISE
+        - AMP_MOD
+        - FREQ_MOD
+        - ARB1
+        - ARB2
+        - ARB3
+        """,
+        validator=strict_discrete_set,
+        map_values=True,
+        values=WAVEFORMS,
+    )
+
+    frequency = Instrument.control(
+        "{ch}\ncf",
+        "{ch}f%.9i",
+        "Control the frequency (Hz) of the channel.",
+        validator=step_validator(1),
+        values=(0, 100000000),
+        set_process=lambda x: x * 100,
+        get_process=lambda x: float(x.strip("cf")) / 100,
+    )
+
+    amplitude = Instrument.setting(
+        "{ch}a%.1f",
+        "Set the amplitude (V) of the channel.",
+        validator=step_validator(0.1),
+        values=(0.1, 20),
+    )
+
+    offset = Instrument.setting(
+        "{ch}o%.1f",
+        "Set the offset (V) of the channel",
+        validator=step_validator(0.1),
+        values=(-10, 10),
+    )
+
+    duty_cycle = Instrument.control(
+        "{ch}\ncd",
+        "{ch}d%i",
+        "Set the duty cycle (percentage) of the channel",
```

Comment:
Also here: Control.

With a single parent class, it would have been one changed line instead of two.